### PR TITLE
Use proper TMPDIR instead of /tmp

### PR DESCRIPTION
**Pull request by segiddins**
\c @alloy @neonichu

### CODE REVIEW

**Comment by segiddins** [comment]
\c @kylef

File: spec/unit/validator_spec.rb [hunk 1]
```diff
@@ -88,7 +88,7 @@ def podspec_path
         validator.quick = true
         validator.stubs(:validate_url)
         validator.validate
-        validator.validation_dir.should.be == Pathname.new('/private/tmp/CocoaPods/Lint')
+        validator.validation_dir.should.be == Pathname.new(ENV['TMPDIR']) + 'CocoaPods/Lint'
```
Comment (neonichu) [inline spec/unit/validator_spec.rb]: Why is the environment used here instead of `Dir.tmpdir`?

File: lib/cocoapods/command/spec/lint.rb [hunk 2]
```diff
@@ -97,7 +97,7 @@ def podspecs_to_lint
         end
 
         def podspecs_tmp_dir
-          Pathname(File.join(Pathname.new('/tmp').realpath, '/CocoaPods/Lint_podspec'))
+          Pathname.new(Dir.tmpdir) + '/CocoaPods/Lint_podspec'
```
Comment (kylef) [inline lib/cocoapods/command/spec/lint.rb]: When using + with Pathname's, there shouldn't be a slash at the start of the second path.

```
irb(main):012:0> Pathname.new(Dir.tmpdir) + '/CocoaPods/Lint_podspec'
=> #<Pathname:/CocoaPods/Lint_podspec>
irb(main):013:0> Pathname.new(Dir.tmpdir) + 'CocoaPods/Lint_podspec'
=> #<Pathname:/var/folders/tj/rgvrw2vj6ygbxvg2kd8f0gl00000gn/T/CocoaPods/Lint_podspec>
```

Comment (segiddins) [inline lib/cocoapods/command/spec/lint.rb]: :+1:

File: spec/unit/validator_spec.rb (hunk 1 above)
Comment (segiddins) [inline spec/unit/validator_spec.rb]: Good point :+1:

**Comment by segiddins** [comment]
Comments addressed. Even fixed a bonus issue while I was at it.

**Pull request merged**